app/views/prison/visits/_cancel_visit.html.erb

```diff
@@ -1,26 +1,29 @@
-<%= form_tag cancel_prison_visit_path(@visit, locale: 'en'), class: 'form' do -%>
+<%= form_for [:prison, @visit, @visit.cancellation], class: 'form' do |f| -%>
+  <%= f.hidden_field :nomis_cancelled, values: true %>
   <h3 class="heading-medium"><%= t(".title") %></h3>
 
   <div class="grid-row">
     <fieldset>
-      <div class="column-one-half">
-        <% %i[prisoner_vos prisoner_released child_protection_issues slot_unavailable visitor_banned].each do |reason| %>
-          <div class="multiple-choice">
-            <%= check_box_tag 'cancellation_reasons[]', reason, false, id: reason %>
-            <%= label_tag reason, t(".#{reason}") %>
+      <%= error_container(f, :reasons, class: 'form-group form-group-error') do %>
+        <% f.object.staff_cancellation_reasons.in_groups_of(5, false).each do |reasons| %>
+          <div class="column-one-half">
+            <%= f.collection_check_boxes :reasons, reasons, :id, :label, include_hidden: false do |b| %>
+              <div class="multiple-choice">
+                <%= b.check_box %>
+                <%= b.label %>
+              </div>
+            <% end %>
           </div>
         <% end %>
-      </div>
-      <div class="column-one-half">
-        <% %i[prisoner_moved prisoner_non_association prisoner_cancelled booked_in_error capacity_issues].each do |reason| %>
-          <div class="multiple-choice">
-            <%= check_box_tag 'cancellation_reasons[]', reason, false, id: reason %>
-            <%= label_tag reason, t(".#{reason}") %>
-          </div>
-        <% end %>
-      </div>
+      <% end %>
+
     </fieldset>
   </div>
+  <div class="panel panel-border-narrow">
+    <p>
+      <%= t(".cancel_to_nomis_warning")%>
+    </p>
+  </div>
```

Comment:
@chrishanes could whisper in some css to fix the alignment and possibly make this message more user friendly?

![screen shot 2017-10-31 at 16 18 18](https://user-images.githubusercontent.com/100796/32235495-899f6ae0-be57-11e7-8f8b-ded5ce7a0742.png)

Reply:
@chrishanes is it the styling consistent with the red bar on the left when there is a validation error?

Is the problem that the form isn't using the form_helpers? If we use them I think we'll be able to remove the global config on `field_error_proc`

Reply:
We are using the form helper `#error_container`, which btw could/should be used by field_error_proc as it is the case in the form_helper_gov_uk gem (or any other form libraries...) 

On another side not we are using error_container around pretty much every field, in the process form which we absolutely don't need since we never display validation around a specific field (except the date for the renewal)

Reply:
👍 we still should have the card for using the gem somewhere